### PR TITLE
Quizzes tests

### PR DESCRIPTION
**Pull request by ndrsszsz**
[Bitrise - Canvas iOS - Student - E2ETests](https://app.bitrise.io/build/41651cb2-eb18-4f4a-8604-46d1f5f02891)

- NewQuizzesHelper
- NewQuizzes extension for DataSeeder
- Updated ModuleTests
- QuizzesHelper
- QuizzesTests with 2 detailed test case (1 for checking details, 1 for taking a quiz)
- FeatureFlag extension for DataSeeder (not used currently, but might be useful in the future so I kept it)
- Some other minor changes/improvements

### CODE REVIEW

**Comment by inst-danger** [comment]
<details><summary>Student Build QR Code:</summary>

</details>

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Please add a release note. If no release note is wanted, use `none`. Example: `release note: Fixed a bug that prevented users from enjoying the app.`
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  Please add which apps this change affects. Example: `affects: Teacher, Student` or `affects: none`
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  Please add a reference to a jira ticket. For example: `refs: MBL-10023`
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>Build failed, skipping coverage check</td>
    </tr>
  </tbody>
</table>



[MBL-16860](https://instructure.atlassian.net/browse/MBL-16860)

<details><summary>:x: Swift lint</summary>


    ❌ /Users/vagrant/git/TestsFoundation/TestsFoundation/DataSeeding/NewQuizzes/CreateDSNewQuizRequest.swift:52:1: Trailing Newline Violation: Files should have a single trailing newline (trailing_newline)
    
    
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 4e3da0a15165e0db918278c35609ccc874848392
</p>

File: Student/StudentE2ETests/Quizzes/QuizzesTests.swift [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        XCTAssertTrue(takeQuizSubmitQuizButton.isVisible)
+
+        takeQuizSubmitQuizButton.tap()
+        takeQuizExitButton = TakeQuizHelper.exitButton.waitToExist()
+        XCTAssertTrue(takeQuizExitButton.isVisible)
+
+        takeQuizExitButton.tap()
+        let detailsStatusLabel = DetailsHelper.statusLabel.waitToExist()
+        XCTAssertTrue(detailsStatusLabel.isVisible)
+        XCTAssertEqual(detailsStatusLabel.label().prefix(9), "Submitted")
```
Comment (vargaat) [inline Student/StudentE2ETests/Quizzes/QuizzesTests.swift:147]:
```suggestion
        XCTAssertTrue(detailsStatusLabel.label().hasPrefix("Submitted"))
```

File: TestsFoundation/TestsFoundation/DataSeeding/FeatureFlags/DataSeeder+DSFeatureFlag.swift [hunk 2]
```diff
@@ -16,26 +16,16 @@
...
+        let request = GetDSFeaturesRequest(courseId: courseId, accountID: accountId)
+        return makeRequest(request)
     }
 
-    func testQuizzesShowEmptyState() {
-        Dashboard.courseCard(id: "262").tap()
-        CourseNavigation.announcements.waitToExist()
-        CourseNavigation.quizzes.waitToVanish()
+    public func setFeatureFlag(courseId: String, feature: DSFeature, state: DSFeatureFlagState) -> DSFeatureFlag {
+        let requestBody = SetDSFeatureFlagRequest.Body.init(state: state)
```
Comment (vargaat) [inline TestsFoundation/TestsFoundation/DataSeeding/FeatureFlags/DataSeeder+DSFeatureFlag.swift:27]:
```suggestion
        let requestBody = SetDSFeatureFlagRequest.Body(state: state)
```

**Review by vargaat — DISMISSED** [review]
Nice to see quiz tests being automated!

**Review by szabinst — APPROVED** [review]

**Review by balintbartok — APPROVED** [review]

**Pull request merged**